### PR TITLE
Rule contribution tests- DynamicQuery

### PR DESCRIPTION
**Pull request by jonlouie**
## Description
* Add unit tests for the DynamicQuery nuget package rules

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

**Review by saikiranakula-amzn — APPROVED** [review]

File: tst/CTA.Rules.Test/RuleContributionTests.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        _resultsDict = new Dictionary<string, TestSolutionAnalysis>
+        {
+            {TargetFramework.DotnetCoreApp31, net31Results},
+            {TargetFramework.Dotnet5, net50Results},
+            {TargetFramework.Dotnet6, net60Results}
+        };
+    }
+    
+    [TestCase(TargetFramework.Dotnet5)]
+    [TestCase(TargetFramework.Dotnet6)]
```
Comment (saikiranakula-amzn) [inline tst/CTA.Rules.Test/RuleContributionTests.cs:43]: what about 3.1?

Comment (jonlouie) [reply tst/CTA.Rules.Test/RuleContributionTests.cs:43]: The rule that was submitted only included .NET 5 and .NET 6, so tests and implementation were of the same scope, though 3.1 was not attempted.

Comment (saikiranakula-amzn) [reply tst/CTA.Rules.Test/RuleContributionTests.cs:43]: since we know this rule is needed can we check compatibilty for 3,1 and add it now as well?

Comment (jonlouie) [reply tst/CTA.Rules.Test/RuleContributionTests.cs:43]: Thanks, Sai. That's a good idea- will validate

File: tst/CTA.Rules.Test/TempRules/system.linq.dynamic.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
+﻿{
```
Comment (Beau-Gosse-dev) [inline tst/CTA.Rules.Test/TempRules/system.linq.dynamic.json:1]: Does this file even get used in the test? Will this file be replaced once the real rule is merged into the porting-assistant-dotnet-datastore repo?

**Review by Beau-Gosse-dev — APPROVED** [review]

File: tst/CTA.Rules.Test/TempRules/system.linq.dynamic.json (hunk 2 above)
Comment (Eruanion) [reply tst/CTA.Rules.Test/TempRules/system.linq.dynamic.json:1]: We will need to remove this file once the rule is merged into the repo. This is how we get around the fact Unit tests are needed before the rule can be merged.

**Review by Eruanion — APPROVED** [review]
We should add 3.1 if it is valid as Sai Kiran mentioned, otherwise looks great thanks.

File: tst/CTA.Rules.Test/TempRules/system.linq.dynamic.json (hunk 2 above)
Comment (Beau-Gosse-dev) [reply tst/CTA.Rules.Test/TempRules/system.linq.dynamic.json:1]: I'll just leave this open as a reminder until we remove it.

**Review by Beau-Gosse-dev — APPROVED** [review]

**Review by Eruanion — APPROVED** [review]

**Review by saikiranakula-amzn — APPROVED** [review]

**Pull request merged**